src/OneExplorer/OneExplorer.ts

```diff
@@ -80,17 +108,22 @@ enum NodeType {
   product,
 }
 
-class Node {
-  type: NodeType;
+abstract class Node {
```

Comment:
(2) Made Node as an abstract class (strategy pattern)

Each type of nodes are made into each classes